package.json

```diff
@@ -3,10 +3,11 @@
   "description": "Project Description",
   "version": "0.0.1",
   "main": "gulpfile.js",
-  "dependencies": {
+  "dependencies": {},
```

Comment:
Lets leave dependencies as a default here, and use devDependencies for nodegit, which is only needed for plugin maintenance where you need to bump version of files modified on a specific release, or things of the sort.

As far as i know, that's the only difference when using dev or normal dependencies, you can use (as noted on the README.md) `node install --production` and you would ignore nodegit when you don't need it.

I thought this would be easier to maintain than keeping 2 separate gulpfiles for different purposes as most of the time it's the same requirements.

Reply:
Thought "dependencies" are used as project dependencies that are required in production env in our case all modules are only required as local dev deps.

Reply:
In node specific projects that's the case, but in WP that's irrelevant since the only environment required is compilation, and we differentiate compilation environments at compilation time, not on package.json.

Reply:
And in case we use this as you intended we can move `semver` package to devDeps and require it in do_bump only

Reply:
True, but technically out of the scope of this PR.

Reply:
Hey @bartech101 can you do the changes so that we're not changing dependencies for devDependencies?

Reply:
[Not specific to this PR]
One thing here: once we are going to use "@wordpress/eslint-plugin" here, why not have "@wordpress/scripts" instead, so we can remove "stylelint", "stylelint-config-wordpress" and "eslint" dependencies as they are all included into the main package?

This is how I have been using it, adding these entries to package.json scripts:
"lint:css": "wp-scripts lint-style",
"lint:css:src": "wp-scripts lint-style '**/source/**/*.scss'",
"lint:js": "wp-scripts lint-js",
"lint:js:src": "wp-scripts lint-js **/source/**/*.js"

Reply:
Lets create a separate PR or issue to discuss the above @pauloiankoski